src/CBot.Modules.NewRelic/API/NewRelicApi.cs

```diff
@@ -49,10 +60,25 @@ public async Task<IEnumerable<Application>> GetFilteredApplicationsSummaryByName
 			return apiResponse.Applications;
 		}
 
-		private HttpClient GetHttpClientWithBaseAddress()
+		private HttpClient GetHttpClientWithBaseAddress(string accountName)
 		{
-			var httpClient = new HttpClient { BaseAddress = new Uri(this.configuration.ApiUrl) };
-			httpClient.DefaultRequestHeaders.Add("X-API-Key", this.configuration.ApiKey);
+			ModuleConfiguration configuration;
+			if (string.IsNullOrWhiteSpace(accountName))
+			{
+				configuration = configurations.First(c => c.IsDefault);
+			}
+			else
+			{
+				configuration = configurations.First(c => c.AccountName.Equals(accountName, StringComparison.InvariantCultureIgnoreCase));
+			}
+
+			if (configuration is null)
+			{
+				logger.LogError($"Failed to find New Relic account by name: {accountName}");
```

Comment:
Should we throw an exception here so we might reference a null object on line 80 when we construct the HttpClient?

Reply:
Yeah, that seems like a good idea. I'll update.